packages/aws-cdk/lib/api/bootstrap/bootstrap-environment.ts

```diff
@@ -131,14 +131,24 @@ export class Bootstrapper {
     const currentKmsKeyId = current.parameters.FileAssetsBucketKmsKeyId;
     const kmsKeyId = params.kmsKeyId ??
       (params.createCustomerMasterKey === true ? CREATE_NEW_KEY :
-        params.createCustomerMasterKey === false || currentKmsKeyId === undefined ? USE_AWS_MANAGED_KEY :
+        params.createCustomerMasterKey === false || currentKmsKeyId === undefined ? USE_DEFAULT_ENCRYPTION :
           undefined);
 
+    let kmsEcrKeyId = USE_DEFAULT_ENCRYPTION;
+    if (params.ecrKeyId) {
+      // key is given
+      kmsEcrKeyId = params.ecrKeyId;
+    } else if (params.ecrKeyId === '') {
```

Comment:
If it is not provided then we should *not* create a key correct?

Reply:
So by default we assume `DEFAULT_ENCRYPTION`.
If somebody provides a specific key id or arn we use that one to encrypt.
If they just apply the 'flag' without a key id we would use the `BOOTSTRAP_KEY` that is why we check if the parameter is an empty string.